### DISCUSSION TITLE
Enable deployment in kubernetes

### DISCUSSION THREAD

**Discussion by jlpedrosa** [jellyfin/jellyfin#11180] [General]
Some user are deploying Jellyfin in Kubernetes through helm charts, a very common way of K8s deployments.

Currently there are multiple open source Helm charts, some of them are specific to NAS servers/distributions, while others are not well maintained. Jellyfin already supports kubernetes and it's the recommended way (officially) to run it in TrueNas, see docs: https://jellyfin.org/docs/general/installation/container#truenas-scale--truecharts 

It would be great if Jellyfin would provide an "official" Helm Chart for non TrueNas environments. I can contribute to the project and creaete a PR for it.

Let me know what you think.

**Comment by adam-mcguinness** [comment]
I am also up to help work on this. Been working on one myself.

**Comment by joshuaboniface** [comment]
I think that is totally reasonable, done the same way as the TrueNas ones. Feel free to proceed, and we'll take it from there.

**Comment by jlpedrosa** [comment]
@adam-mcguinness you want to go for it? Or do you want me to do it. Happy either way.

**Reply by adam-mcguinness** [reply]
I went ahead and forked the Jellyfin repo and added my helm chart to the deployment directory. My experience with Kubernetes is somewhat limited, so feel free to take over. This was also built for my environment utilizing a direct nfs volume mount for files instead of PVCs, and an nginx ingress controller for serving to the internet. 

Added some docs in the README.md for the chart as well.

https://github.com/adam-mcguinness/jellyfin/tree/helm-deployment

**Comment by jlpedrosa** [comment]
@joshuaboniface: I will create a PR with the chart. I assume you will want to take care of CI to publish the charts yourself. I can also jump to help in that one in a later pr too as well as docs.

**Reply by Shadowghost** [reply]
I think it would be cleaner to have the chart in its own repo

**Reply by jlpedrosa** [reply]
Most of the repos have the chart colocated with the code. Also from CI point of view is simpler (new appVersion, new chart version) otherwise you'll have repos talking to each other (or renovate).

Just let me know the outcome and where to do the PR.

**Reply by jlpedrosa** [reply]
@adam-mcguinness @Shadowghost  Let me know what is the outcome if you want it in repo or you guys want another repo for it?

I'll create the PR as soon as you give me the destination!

**Reply by joshuaboniface** [reply]
Sorry I didn't see this ping back in March.

While that may be true for most software, in our case the code is actually split among multiple repositories. The server code (one half of "Jellyfin") is in one repo (this one), the web code is in another (the other half) and all packaging is in its own repo to keep it independent from those other two.

Since this deals primarily with Docker (basically), I'm OK with it going into the Packaging repo (https://github.com/jellyfin/jellyfin-packaging) along with the Dockerfiles. But I would also prefer it be its own repository so it can be managed and tracked separately (e.g. issues, PRs, etc.)

**Reply by jlpedrosa** [reply]
hey @joshuaboniface! no problem! if you create the repo `/jellyfin/jellylin-helm` or what ever you consider I'll create the PR, TBH it simplifies the CI for the component.

**Reply by crobibero** [reply]
@jlpedrosa https://github.com/jellyfin/jellyfin-helm

**Reply by jlpedrosa** [reply]
@crobibero && @joshuaboniface  I have the PR ready, but there's no a single commit it that repo, it doesn't allow me to fork it create PRs and I don't have write access.

**Reply by crobibero** [reply]
@jlpedrosa hah well that is fun to know! I just pushed an empty README. You should be good to go now

**Reply by jlpedrosa** [reply]
@crobibero, @joshuaboniface  https://github.com/jellyfin/jellyfin-helm/pull/2
